src/main.rs

```diff
@@ -39,6 +39,19 @@ struct Opts {
     )]
     ignore_file: String,
 
+    #[options(no_short, help = "Exclude files with this tag from the export")]
+    skip_tags: Vec<String>,
+
+    #[options(no_short, help = "Only include files with this tag in the export")]
+    only_tags: Vec<String>,
+
+    #[options(
+        no_short,
+        help = "Exclude files with this flag in the frontmatter from the export",
+        default = "private"
+    )]
+    ignore_frontmatter_flag: String,
```

Comment:
I'm not feeling too sure about keeping this in.

`--skip-tags private` effectively achieves the same behavior, but in a more generic and flexible way. Are there any use-cases where setting `private: true` in frontmatter has clearly better end-user ergonomics over asking people to just set a tag with value `private`?

There's already multiple ways to adjust what obsidian-export selects or ignores (exclude files, `--start-at`, and now these new options) and while I'm obviously not against offering flexibility, I would like to avoid this growing into having to support tens of options that do almost the same thing.

Even more so because unlike `--skip-tags` and `--only-tags` that are optional/no-op when unspecified, this option is forced on users now.

Are you okay with settling just on `--skip-tags` and `--only-tags`, dropping this one?

Reply:
I included it since I was already halfway there, I don't have any problems with dropping it.
do you have any advice on programmatically changing the frontmatter from `private: true` to a tag?

Reply:
> I included it since I was already halfway there, I don't have any problems with dropping it.

Great! Then let's remove that code from the implementation and settle on just the `--skip-tags` and `--only-tags`.

> do you have any advice on programmatically changing the frontmatter from `private: true` to a tag?

My go-to for this tends to be one-off Python scripts, because it's so easy to quickly iterate on something that works, even if it's ugly or inefficient :see_no_evil: 

As it happens, you're in luck because I have an example of such a script which I used to update/set some date-related frontmatter attributes, that should get you 90% of the way there already. Uploaded it as a gist at https://gist.github.com/zoni/d3f6a803f552cc946d3bc00447d3e051 (requires [python-frontmatter](https://pypi.org/project/python-frontmatter/))

You'd just have to change `update_frontmatter` so it looks something like (untested):

```python
def update_frontmatter(path):
    doc = frontmatter.load(path)
    tags = doc.get("tags", [])
    private = doc.get("private", False)

    if private:
        tags += ["private"]
    doc["tags"] = sorted(set(tags))

    path.write_text(frontmatter.dumps(doc), encoding="utf-8")
```